stages/org.osbuild.coreos.live-artifacts.mono

```diff
@@ -792,29 +792,25 @@ def mk_paths(workdir):
     return {
         "live.iso": os.path.join(workdir, "live.iso"),
         "efi": os.path.join(workdir, "efi"),
-        "iso": {
-            "root": os.path.join(workdir, "iso"),
-            "coreos": os.path.join(workdir, "iso", "coreos"),
-            "isolinux": os.path.join(workdir, "iso", "isolinux"),
-            "images": os.path.join(workdir, "iso", "images"),
-            "pxeboot": os.path.join(workdir, "iso", "images", "pxeboot"),
-            "initramfs": os.path.join(workdir, "iso", "images", "pxeboot", "initrd.img"),
-        },
+        "iso": os.path.join(workdir, "iso"),
+        "iso/coreos": os.path.join(workdir, "iso/coreos"),
+        "iso/images": os.path.join(workdir, "iso/images"),
+        "iso/images/pxeboot": os.path.join(workdir, "iso/images/pxeboot"),
+        "iso/images/pxeboot/initrd.img": os.path.join(workdir, "iso/images/pxeboot/initrd.img"),
+        "iso/isolinux": os.path.join(workdir, "iso/isolinux"),
```

Comment:
I have to question the usefulness of a dictionary where the values are (more or less) equivalent to the keys.  The dict is basically a precomputed `os.path.join(workdir, key)` now which is fine, but I'm not sure what it provides other than maybe catching typos in paths that don't show up in the `mk_workdirs()` function.

The previous dictionary implementation had one advantage: changing a path only required a change in one place to the value.  I realise none of these paths are likely to change, so maybe that's a nonsense benefit.

The original implementation, which was one variable name per path, had the same advantage but also guarded against mistyping and had the benefit of things being caught by static checkers.

As for the second implementation, one function per path, that had the same benefits of the original, with the downside of needing to make function calls all the time but the upside of having to only carry the workdir around into function arguments.

The best solution might be a class that would work a bit like (thanks @mvo5 for this):
```python
class Paths:
  def __init__(self, workdir):
    self._workdir = workdir

  @property
  def images(self):
    # alternatively, set self._images in __init__ and return it here.
    return os.path.join(self._workdir, "images")

paths = Paths(workdir)
...
images = paths.images
```
which would benefit from static checks, intelligent refactoring, a single call to `os.path.join()` if done in `__init__`, and a single `paths` object being passed around.

At the end of the day, it's your call though.  If you tell me to stick with this dict, then that's fine.  We probably spent too much time on this tiny detail already anyway.

Reply:
> I have to question the usefulness of a dictionary where the values are (more or less) equivalent to the keys.

I agree. For my brain this was just the way that helped make everything more concrete. It does have the drawbacks you mention, though. 

> At the end of the day, it's your call though. If you tell me to stick with this dict, then that's fine. We probably spent too much time on this tiny detail already anyway.

Yeah. I think maybe let's stick with this (in the interest of at least getting this part done) and open a followup PR to discuss the merits of a better way to pass the path information around. I think @mvo5 and I will probably have some followups for tests anyway. WDYT?

Reply:
Yup.  Resolving the thread.